### PR TITLE
gtk: Cleanup template related functions

### LINKED ISSUE #1189: Improve bindings for gtk::Widget::dispose_template
Currently one has to call
```rust
       fn dispose(&self) {
            self.obj().dispose_template(Self::Type::static_type());
        }
```
but this could be turn into
```rust
        fn dispose(&self) {
            self.dispose_template();
        }
```

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
Fixes #1189

### CODE REVIEW

**Comment by sdroege** [comment]
That's of course not backwards compatible so don't backport that :)

**Comment by bilelmoussaoui** [comment]
> That's of course not backwards compatible so don't backport that :)

yes, that is why i didn't add the corresponding label

File: gtk4/src/auto/widget.rs [hunk 1]
```diff
@@ -372,10 +372,6 @@ pub trait WidgetExt: 'static {
     #[doc(alias = "get_style_context")]
     fn style_context(&self) -> StyleContext;
 
-    #[doc(alias = "gtk_widget_get_template_child")]
-    #[doc(alias = "get_template_child")]
-    fn template_child(&self, widget_type: glib::types::Type, name: &str) -> glib::Object;
```
Comment (sdroege) [inline gtk4/src/auto/widget.rs:382]: Doesn't hurt to have it, does it?

Comment (bilelmoussaoui) [reply gtk4/src/auto/widget.rs:382]: There is no point in having it, you can retrieve child widgets with `#[template_child]`. This is probably a helper for languages like Python

**Pull request merged**